servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```
@@ -1117,11 +1250,89 @@ public void reportLostFile(long fileId) throws FileDoesNotExistException {
+      return fileId;
+    } catch (IOException e) {
+      LOG.error(ExceptionUtils.getStackTrace(e));
+      throw new TachyonException(e.getMessage());
     }
   }
 
+  public boolean mount(TachyonURI tachyonPath, TachyonURI ufsPath)
+      throws FileAlreadyExistException, InvalidPathException, IOException {
+    synchronized (mInodeTree) {
+      mkdirs(tachyonPath, true);
+      if (mountInternal(tachyonPath, ufsPath)) {
+        writeJournalEntry(new AddMountPointEntry(tachyonPath, ufsPath));
+        flushJournal();
+        return true;
+      }
+    }
+    return false;
+  }
+
+  void mountFromEntry(AddMountPointEntry entry) {
+    TachyonURI tachyonPath = entry.getTachyonPath();
+    TachyonURI ufsPath = entry.getUfsPath();
+    try {
+      if (!mountInternal(tachyonPath, ufsPath)) {
+        LOG.error("Failed to mount " + ufsPath + " at " + tachyonPath);
+      }
+    } catch (InvalidPathException e) {
+      LOG.error("Failed to mount " + ufsPath + " at " + tachyonPath);
+    }
+  }
+
+  boolean mountInternal(TachyonURI tachyonPath, TachyonURI ufsPath) throws InvalidPathException {
+    return mMountTable.add(tachyonPath, ufsPath);
+  }
+
+  public boolean unmount(TachyonURI tachyonPath) throws FileDoesNotExistException,
+      InvalidPathException, IOException {
+    synchronized (mInodeTree) {
+      if (unmountInternal(tachyonPath)) {
+        Inode inode = mInodeTree.getInodeByPath(tachyonPath);
+        deleteFile(inode.getId(), true);
+        writeJournalEntry(new DeleteMountPointEntry(tachyonPath));
+        flushJournal();
+        return true;
+      }
+    }
+    return false;
+  }
+
+  void unmountFromEntry(DeleteMountPointEntry entry) {
+    TachyonURI tachyonPath = entry.getTachyonPath();
+    try {
+      if (!unmountInternal(tachyonPath)) {
+        LOG.error("Failed to unmount " + tachyonPath);
+      }
+    } catch (InvalidPathException e) {
+      LOG.error("Failed to unmount " + tachyonPath);
```

Are we purposely swallowing this exception?